src/test/run-pass/vec-tail-matching.rs

```diff
@@ -31,8 +31,8 @@ pub fn main() {
                     unreachable!();
                 }
                 [Foo { string: ref a }, Foo { string: ref b }] => {
-                    assert_eq!("bar", a.as_slice().slice(0, a.len()));
-                    assert_eq!("baz", b.as_slice().slice(0, b.len()));
+                    assert_eq!("bar", &a[0..a.len()]);
+                    assert_eq!("baz", &b[0..b.len()]);
```

Comment:
`..x` is preferred. But, I have mapped the methods to their corresponding slice notations without performing any simplification to retain the meaning of the tests (even if they don't make sense to me).